pkg/controller/node/status.go

```diff
@@ -221,13 +222,26 @@ func isNodeReady(node *corev1.Node) bool {
 	return checkNodeReady(node) == nil
 }
 
+func isNodeOnHold(node *corev1.Node) bool {
```

Comment:
Could we add comment here explaining the what/why?  like from your pr description?